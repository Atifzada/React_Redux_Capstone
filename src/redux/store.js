import { configureStore, combineReducers } from '@reduxjs/toolkit';
import cryptoReducer from './cryptoSlice';
import detailsReducer from './detailSlice';

const reducer = combineReducers({
  crypto: cryptoReducer,
  cryptoDetails: detailsReducer,
});

const store = configureStore({ reducer });

export default store;
