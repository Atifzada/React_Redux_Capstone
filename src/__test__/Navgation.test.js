import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import Nav from '../components/Navgation';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Navigation page testing', () => {
  test('Renders nav accuratly ', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <Nav />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('renders correctly logo and text ', () => {
    render(
      <Router>
        <Nav />
      </Router>,
    );
    const logo = screen.getByAltText('Logo');
    const text = screen.getByText('(MCT)');
    expect(logo).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });
});
