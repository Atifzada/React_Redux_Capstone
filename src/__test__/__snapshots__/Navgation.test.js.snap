// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation page testing Renders nav accuratly  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="myflex darker-bg"
      >
        <div
          class="myflex navBox"
        >
          <div
            aria-hidden="true"
            class="myflex back"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M328 112L184 256l144 144"
                fill="none"
                stroke-linecap="square"
                stroke-miterlimit="10"
                stroke-width="48"
              />
            </svg>
          </div>
          <div
            class="myflex navTitle"
          >
            <a
              href="/"
            >
              <img
                alt="Logo"
                class="navLogo"
                src="logo.png"
              />
              (MCT)
            </a>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="myflex darker-bg"
    >
      <div
        class="myflex navBox"
      >
        <div
          aria-hidden="true"
          class="myflex back"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M328 112L184 256l144 144"
              fill="none"
              stroke-linecap="square"
              stroke-miterlimit="10"
              stroke-width="48"
            />
          </svg>
        </div>
        <div
          class="myflex navTitle"
        >
          <a
            href="/"
          >
            <img
              alt="Logo"
              class="navLogo"
              src="logo.png"
            />
            (MCT)
          </a>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
